src/systems/systems/src/landing_gear/mod.rs

```diff
@@ -744,6 +756,95 @@ impl LandingGearControlInterfaceUnit {
     fn status(&self) -> LgciuStatus {
         self.status
     }
+
+    fn discrete_word_1(&self) -> Arinc429Word<f32> {
+        if !self.is_powered {
+            Arinc429Word::new(0., SignStatus::FailureWarning)
+        } else {
+            let mut word = Arinc429Word::new(0., SignStatus::NormalOperation);
+            word.set_bit(
+                11,
+                !self.sensor_inputs.left_gear_up_and_locked && !self.gear_handle_is_down(),
+            );
+            word.set_bit(
+                12,
+                !self.sensor_inputs.right_gear_up_and_locked && !self.gear_handle_is_down(),
+            );
+            word.set_bit(
+                13,
+                !self.sensor_inputs.nose_gear_up_and_locked && !self.gear_handle_is_down(),
+            );
+            word.set_bit(
+                14,
+                !self.sensor_inputs.left_gear_down_and_locked && self.gear_handle_is_down(),
+            );
+            word.set_bit(
+                15,
+                !self.sensor_inputs.right_gear_down_and_locked && self.gear_handle_is_down(),
+            );
+            word.set_bit(
+                16,
+                !self.sensor_inputs.nose_gear_down_and_locked && self.gear_handle_is_down(),
+            );
+            word.set_bit(17, !self.sensor_inputs.left_door_up_and_locked);
+            word.set_bit(18, !self.sensor_inputs.right_door_up_and_locked);
+            word.set_bit(19, !self.sensor_inputs.nose_door_up_and_locked);
+            // The conditions for bits 20, 21 and 22 are not implemented, so they are set to false for now.
+            word.set_bit(20, false);
+            word.set_bit(21, false);
+            word.set_bit(22, false);
+            word.set_bit(23, self.sensor_inputs.downlock_state(GearWheel::LEFT));
+            word.set_bit(24, self.sensor_inputs.downlock_state(GearWheel::RIGHT));
+            word.set_bit(25, self.sensor_inputs.downlock_state(GearWheel::NOSE));
+            word.set_bit(26, self.sensor_inputs.left_gear_compressed(true));
+            word.set_bit(27, self.sensor_inputs.right_gear_compressed(true));
+            word.set_bit(28, self.sensor_inputs.nose_gear_compressed(true));
+            word.set_bit(29, self.gear_handle_is_down());
+
+            word
+        }
+    }
+
+    fn discrete_word_2(&self) -> Arinc429Word<f32> {
+        if !self.is_powered {
+            Arinc429Word::new(0., SignStatus::FailureWarning)
+        } else {
+            let mut word = Arinc429Word::new(0., SignStatus::NormalOperation);
```

Comment:
**suggestion**: Consider creating another constructor function which doesn't take a value, as it seems these usages always set the value to `0.`.

```rust
impl Arinc429Word<u32> {
    pub fn bits(ssm: SignStatus) -> Self {
        Self { value: 0., ssm }
    }
}
```

Reply:
I think it's better to keep it more explicit, so you can directly see it initializes to 0 and don't have to look up what the constructor does. Also, IMO, it wouldn't really save much to write anyways